File: backend-systems-lab-stage-1/src/main/java/pl/jakubtworek/backend_systems_lab_stage_1/block_a/kotlin/basics/nullsafety/java/Address.java
package pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.basics.nullsafety.java;

// A simple Address class used by UserProfile.
public class Address {

    // City may be null, but this is not visible in the type system.
    private final String city;

    public Address(String city) {
        this.city = city;
    }

    public String getCity() {
        return city;
    }
}

File: backend-systems-lab-stage-1/src/main/java/pl/jakubtworek/backend_systems_lab_stage_1/block_a/kotlin/basics/nullsafety/kotlin/Address.kt
package pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.basics.nullsafety.kotlin

// A simple Address class used by UserProfile.
data class Address(

    // Non-nullable city.
    // Kotlin does not allow null here unless we explicitly write String?.
    val city: String
)